src/common/motor.rs

```diff
@@ -35,47 +48,56 @@ pub(crate) struct MotorConfig {
 pub struct FakeMotor {
     pos: f64,
     power: f64,
+    max_rpm: f64,
 }
 
-impl TryFrom<Kind> for MotorConfig {
+impl TryFrom<Kind> for MotorPinsConfig {
     type Error = AttributeError;
     fn try_from(value: Kind) -> Result<Self, Self::Error> {
-        let mut motor = MotorConfig::default();
+        let mut motor = MotorPinsConfig::default();
         match value {
             Kind::StructValueStatic(v) => {
-                if !v.contains_key("pwm") {
-                    return Err(AttributeError::KeyNotFound);
-                }
-                motor.pwm = v.get("pwm").unwrap().try_into()?;
-                if v.contains_key("a") {
-                    motor.a = Some(v.get("a").unwrap().try_into()?);
-                }
-                if v.contains_key("b") {
-                    motor.b = Some(v.get("b").unwrap().try_into()?);
-                }
+                motor.pwm = v
+                    .get("pwm")
+                    .ok_or(AttributeError::KeyNotFound("pwm".to_string()))?
+                    .try_into()?;
+                motor.a = Some(
+                    v.get("a")
+                        .ok_or(AttributeError::KeyNotFound("a".to_string()))?
```

Comment:
I don't think we should be returning an error in the absence of A or B some motor controller will have logics different from  A/B/PWM